src/devices/bus/epson_qx/ide.cpp

```diff
@@ -0,0 +1,115 @@
+// license:BSD-3-Clause
+// copyright-holders:Brian Johnson
+/*******************************************************************
+ *
+ * IDE Compact Flash Adapter
+ *
+ * This implements a Compact flash adapter expansion card for QX-10
+ * systems. The boot disk necessary to support this card is listed below.
+ *
+ * Disk Image: https://github.com/brijohn/qx10/raw/master/diskimages/qx10cf.imd
+ * Board Design: https://github.com/brijohn/qx10/tree/master/kicad/cf_adapter
+ *
+ *******************************************************************/
+
+#include "emu.h"
+#include "ide.h"
+
+
+//**************************************************************************
+//  EPSON IDE DEVICE
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(EPSON_QX_OPTION_IDE, bus::epson_qx::ide_device, "epson_qx_option_ide", "Epson QX-10 Compact Flash Adapter")
+
+namespace bus::epson_qx {
+
+INPUT_PORTS_START( IDE )
```

Comment:
Please use `snake_case` for the name of the input port definitions, and make it static or put it in an anonymous namespace to stop the symbol leaking.